src/goto-instrument/code_contracts.cpp

```diff
@@ -370,25 +402,74 @@ void code_contractst::add_contract_check(
   dest.destructive_insert(dest.instructions.begin(), check);
 }
 
-void code_contractst::operator()()
+bool code_contractst::replace_calls(
+  const std::list<std::string> &funs_to_replace)
 {
-  Forall_goto_functions(it, goto_functions)
-    code_contracts(it->second);
-
-  goto_functionst::function_mapt::iterator i_it=
-    goto_functions.function_map.find(INITIALIZE_FUNCTION);
-  assert(i_it!=goto_functions.function_map.end());
+  bool fail = false;
+  Forall_goto_functions(fit, goto_functions)
+  {
+    Forall_goto_program_instructions(it, fit->second.body)
+    {
+      if(it->is_function_call())
+      {
+        const code_function_callt &call = it->get_function_call();
+
+        // TODO we don't handle function pointers
+        if(call.function().id() != ID_symbol)
+          continue;
+
+        const irep_idt &fun_name =
+          to_symbol_expr(call.function()).get_identifier();
+        auto found = std::find(
```

Comment:
Rather than using a std::list for funs_to_replace, is there any efficiency gain to use a std::set? Just wondering how large you expect funs_to_replace to be, given that you'll be doing a search through it for every call site in the program?